src/services/big-query-client.ts

```diff
@@ -0,0 +1,41 @@
+import { ResultWithoutExpectedErrors } from '../doppler-types';
+import { ResponseCache } from './../utils';
+import { GenericClient } from './generic-client';
+
+export interface EmailList {
+  emails: String[];
+}
+
+export interface BigQueryClient {
+  getEmailsData(idUser: String): Promise<ResultWithoutExpectedErrors<EmailList>>;
+}
+
+export class HttpBigQueryClient extends GenericClient implements BigQueryClient {
+  private responseCache = new ResponseCache();
+
+  private mapEmails(response: any): EmailList {
+    return {
+      emails: response.emails,
+    };
+  }
+  public async getEmailsData(idUser: String): Promise<ResultWithoutExpectedErrors<EmailList>> {
+    try {
+      const { jwtToken } = super.getApiConnectionData();
+      const response = await this.axios.request({
```

Comment:
Why do you use `super` for using `getApiConnectionData` and `this` for using `axios`?

Would it work?

```suggestion
      const { jwtToken } = this.getApiConnectionData();
      const response = await this.axios.request({
```

Reply:
in this case is because the method implementation exist on the father class, with this it works but it is not readable that the method is implemented in the parent...

Reply:
But `this` allows to override it, and it is desirable, right?

Reply:
suggested accepted and applied